### PR TITLE
Retire use of generic blob publish

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Retire use of the public-blob.yml from build tools repo.
- Add template package for android.

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-android-release.yml [hunk 1]
```diff
@@ -114,13 +114,10 @@ stages:
                 runOnce:
                   deploy:
                     steps:
-                      - checkout: azure-sdk-build-tools
-                        path: azure-sdk-build-tools
-                      - template: /tools/generic-blob-upload/publish-blobs.yml@azure-sdk-build-tools
+                      - checkout: self
```
Comment (benbp) [inline eng/pipelines/templates/stages/archetype-android-release.yml:117]: Do we need this checkout line anymore if we're not doing multiple repo checkouts?

File: sdk/template/ci.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+resources:
+  repositories:
+    - repository: azure-sdk-build-tools
```
Comment (benbp) [inline sdk/template/ci.yml:3]: Is this still necessary?

File: sdk/template/ci.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      - main
+      - feature/*
+      - hotfix/*
+      - release/*
+  paths:
+    include:
+      - sdk/template/
+
+stages:
+  - template: ../../eng/pipelines/templates/stages/archetype-sdk-client.yml
```
Comment (benbp) [inline sdk/template/ci.yml:36]:
```suggestion
  - template: /eng/pipelines/templates/stages/archetype-sdk-client.yml
```

File: sdk/template/ci.yml (hunk 2 above)
Comment (weshaggard) [reply sdk/template/ci.yml:3]: hopefully not.

Comment (chidozieononiwu) [reply sdk/template/ci.yml:3]: Yes, it's still used for Publishing and GPG signing

File: sdk/template/ci.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      - main
+      - feature/*
+      - hotfix/*
+      - release/*
+  paths:
+    include:
+      - sdk/template/
+
+extends:
+  template: ../../eng/pipelines/templates/stages/archetype-sdk-client.yml
```
Comment (weshaggard) [inline sdk/template/ci.yml:24]:
```suggestion
  template: /eng/pipelines/templates/stages/archetype-sdk-client.yml
```

File: eng/pipelines/templates/stages/archetype-android-release.yml [hunk 5]
```diff
@@ -84,14 +101,12 @@ stages:
               dependsOn: TagRepository
 
               pool:
-                vmImage: vs2017-win2016
+                vmImage: windows-2019
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-android-release.yml:104]: We should use the machine pool here as well.

**Comment by chidozieononiwu** [comment]
Closing this for now

**Comment by weshaggard** [comment]
@chidozieononiwu is there a tracking issue for this work? If so could you please assign it to @JimSuplizio so that we can try and clean-up the android release pipelines.

**Pull request closed without merging**